torchtext/experimental/datasets/raw/common.py

```diff
@@ -14,16 +14,18 @@ class RawTextIterableDataset(torch.utils.data.IterableDataset):
     """Defines an abstraction for raw text iterable datasets.
     """
 
-    def __init__(self, name, full_num_lines, iterator):
+    def __init__(self, name, full_num_lines, iterator, offset=0, stride=1):
         """Initiate text-classification dataset.
         """
         super(RawTextIterableDataset, self).__init__()
         self.name = name
         self.full_num_lines = full_num_lines
         self._iterator = iterator
         self.has_setup = False
-        self.start = 0
+        self.start = offset
         self.num_lines = None
+        self.setup_iter(start=offset, num_lines=full_num_lines - offset)
+        self.stride = stride
 
     def setup_iter(self, start=0, num_lines=None):
```

Comment:
I think it makes sense to inline this code and remove the setup_iter method until we need it again.